lib/generator.js

```diff
@@ -385,9 +392,24 @@ class Generator {
 
       if (isFileSystemPath(this.templateName)) log.debug(logMessage.NPM_INSTALL_TRIGGER);
 
-      const arb = new Arborist({
-        path: ROOT_DIR
-      });
+      const arbOptions = {
+        path: ROOT_DIR,
+      };
+      if (this.registry) {
+        if (this.registry.url) {
+          arbOptions.registry = this.registry.url;
+        }
+        if (this.registry.username) {
+          arbOptions.username = this.registry.username;
+        }
+        if (this.registry.password) {
+          arbOptions.password = this.registry.password;
+        }
+        if (this.registry.token) {
+          arbOptions.token = this.registry.token;
+        }
+      }
```

Comment:
```suggestion
      if (this.registry) {
        if (this.registry.url) arbOptions.registry = this.registry.url;
        if (this.registry.username) arbOptions.username = this.registry.username;
        if (this.registry.password) arbOptions.password = this.registry.password;
        if (this.registry.token) arbOptions.token = this.registry.token;
      }
```

do you mind if we do it this way? it is a shorter and less scary `if` list without so many curly braces.

Reply:
change this one too